libraries/core_libs/consensus/src/dag/dag.cpp

```diff
@@ -584,25 +596,89 @@ uint DagManager::setDagBlockOrder(blk_hash_t const &new_anchor, uint64_t period,
   assert(dag_order_set.count(new_anchor));
   addToDag(new_anchor, blk_hash_t(), vec_blk_t(), 0, true);
 
+  std::pair<std::unordered_map<blk_hash_t, std::shared_ptr<DagBlock>>, std::vector<trx_hash_t>>
+      expired_dag_blocks_to_remove;
```

Comment:
This whole std::pair seems overcomplicated and could be simplified, which would also improved readability:

Instead of 1 pair, you could declre two separate containers with proper names:
```
std::unordered_map<blk_hash_t, std::shared_ptr<DagBlock>> ct1;
std::vector<trx_hash_t> ct2;
```

Read comments below, which are related

Reply:
Refactored